contracts/pool_stable/src/contract.rs

```diff
@@ -1003,7 +1074,11 @@ pub fn compute_swap(
         0
     };
     let return_amount = convert_u128_to_i128(return_amount);
-    let commission_amount = return_amount * commission_rate;
+    let commission_amount = convert_u128_to_i128(
+        Decimal256::new(env, convert_i128_to_u128(return_amount))
+            .mul(env, &commission_rate)
+            .to_u128_with_precision(greatest_precision as i32),
+    );
```

Comment:
```suggestion
    let commission_amount = commission_rate.mul_u128(return_amount)
```
more or less like that